### PR TITLE
Replace AR with ActiveRecord to make it more readable [ci skip]

### PR DESCRIPTION
**Pull request by arvindmehra**

### CODE REVIEW

**Comment by senny** [comment]
@arvindmehra I agree that it's good to use "Active Record" and in the case of code `ActiveRecord::`. Changing AR for internal comments targeting ad framework authors is not helping though. I'll mark the changes that we can apply, please revert the others to keep the patch focused.

File: activerecord/lib/active_record/inheritance.rb [hunk 1]
```diff
@@ -82,7 +82,7 @@ def finder_needs_type_condition? #:nodoc:
       # Returns the class descending directly from ActiveRecord::Base, or
       # an abstract class, if any, in the inheritance hierarchy.
       #
-      # If A extends AR::Base, A.base_class will return A. If B descends from A
+      # If A extends ActiveRecord::Base, A.base_class will return A. If B descends from A
```
Comment (senny) [inline activerecord/lib/active_record/inheritance.rb]: :+1:

File: activerecord/lib/active_record/model_schema.rb [hunk 2]
```diff
@@ -240,7 +240,7 @@ def type_for_attribute(attr_name) # :nodoc:
       end
 
       # Returns a hash where the keys are column names and the values are
-      # default values when instantiating the AR object for this table.
+      # default values when instantiating the ActiveRecord object for this table.
```
Comment (senny) [inline activerecord/lib/active_record/model_schema.rb]: "Active Record" not "ActiveRecord"

File: activerecord/lib/active_record/railties/databases.rake [hunk 3]
```diff
@@ -229,7 +229,7 @@ db_namespace = namespace :db do
   end
 
   namespace :schema do
-    desc 'Create a db/schema.rb file that is portable against any DB supported by AR'
+    desc 'Create a db/schema.rb file that is portable against any DB supported by ActiveRecord'
```
Comment (senny) [inline activerecord/lib/active_record/railties/databases.rake]: "Active Record" not "ActiveRecord"

File: guides/source/3_2_release_notes.md [hunk 4]
```diff
@@ -327,7 +327,7 @@ Active Record
 
 * Implemented `ActiveRecord::Relation#explain`.
 
-* Implements `AR::Base.silence_auto_explain` which allows the user to selectively disable automatic EXPLAINs within a block.
+* Implements `ActiveRecord::Base.silence_auto_explain` which allows the user to selectively disable automatic EXPLAINs within a block.
```
Comment (senny) [inline guides/source/3_2_release_notes.md]: :+1:

File: railties/lib/rails/tasks/engine.rake [hunk 5]
```diff
@@ -40,7 +40,7 @@ namespace :db do
   desc "Rolls the schema back to the previous version (specify steps w/ STEP=n)."
   app_task "rollback"
 
-  desc "Create a db/schema.rb file that can be portably used against any DB supported by AR"
+  desc "Create a db/schema.rb file that can be portably used against any DB supported by ActiveRecord"
```
Comment (senny) [inline railties/lib/rails/tasks/engine.rake]: "Active Record" not "ActiveRecord"

**Comment by senny** [comment]
@arvindmehra I marked the valid occurrences with a comment. Please revert the others.

**Comment by senny** [comment]
Also make sure to add `[ci skip]` to the commit message. This prevents our CI-Server from running.

File: guides/source/3_2_release_notes.md (hunk 4 above)
Comment (arvindmehra) [inline guides/source/3_2_release_notes.md]: @senny thanks for review. I have committed changes that i think should really be helpful to documentation of framework. And thanks for the [ci skip] tip will this in mind.

**Comment by arvindmehra** [comment]
I have reverted others. Please review.

File: activerecord/lib/active_record/railties/databases.rake (hunk 3 above)
Comment (kaspth) [inline activerecord/lib/active_record/railties/databases.rake]: @arvindmehra this is valid case. You just need to put a space in there. That's what @senny meant. The same goes for that other place where there's a space missing.

**Comment by senny** [comment]
@arvindmehra the ones I marked with "Active Record" not "ActiveRecord" are fine two. You just need to add a space in between. Can you reapply those?

**Comment by arvindmehra** [comment]
@senny I have reapplied the changes.

**Comment by senny** [comment]
@arvindmehra you missed:
* https://github.com/rails/rails/pull/21530#discussion_r38874197
* https://github.com/rails/rails/pull/21530#discussion_r38868963

**Comment by arvindmehra** [comment]
@senny  I think have applied.
https://github.com/rails/rails/pull/21530/files#diff-6c8fd220c92c2b25aa29891b59d00c04R243

https://github.com/rails/rails/pull/21530/files#diff-4f92b6693e7bdfd5cf90ec85ccf6ede5R43

**Comment by senny** [comment]
@arvindmehra https://github.com/rails/rails/pull/21530#discussion_r38874197 is still missing. It's in `databases.rake`

**Comment by arvindmehra** [comment]
@senny I checked, its outdated because somehow its already there. https://github.com/rails/rails/blob/master/activerecord/lib/active_record/railties/databases.rake#L232

**Comment by senny** [comment]
:+1: thanks.

**Pull request merged**